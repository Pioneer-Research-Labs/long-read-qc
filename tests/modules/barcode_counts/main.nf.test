nextflow_process {

    name "Test Process barcode_counts"
    script "modules/barcode_counts/main.nf"
    process "barcode_counts"

    test("Should generate barcode counts"){

        when {
            params {
                outdir = outputDir
            }
            process {
                """
                input[0] = Channel.fromList([ tuple([ id:'Gateway_EmptyBarcodedLib1_20241217',
                genome:'H_elongata'],
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/barcodes.fasta'))])
                """
            }
        }

        then {
            assert process.success
            assert snapshot("$outputDir/Gateway_EmptyBarcodedLib1_20241217/barcode_counts.tsv").match()
        }
    }
}

