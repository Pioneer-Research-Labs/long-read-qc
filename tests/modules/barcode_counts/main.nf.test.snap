{
    "Should generate barcode counts": {
        "content": [
            "barcode_counts.tsv:md5,123f794003cc447241814baf03f3f6e4"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-15T11:52:54.051047"
    }
}